### PR TITLE
Separate core requirements v2

### LINKED ISSUE #48: Optional dependancies causing downstream difficulties
The optional dependancies are quite heavy compared to the core Starlette package. It would be good to remove these to keep this lightweight.

I could also build out a separate conda package called `starlette-core-feedstock` which is analogous to the Dask ecosystem.

### PR DESCRIPTION
**Pull request by dgasmith**
This splits the Starlette package into "core" and non-core components so that all of the extra dependancies do not ride along with the package. This is very useful to prevent these unused dependancies.

Closes #48. 

Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by dgasmith** [comment]
Looks like we should be able to move the script/ build number back into a global header instead of a per-output header as well after looking at: https://github.com/bioconda/bioconda-recipes/commit/5f7a62163b38aa9f4aaec79d7685a3a0b9c6e9a9

**Comment by dgasmith** [comment]
@conda-forge-admin, please rerender

**Comment by dgasmith** [comment]
This seems to work, biggest concern at the moment is that `script: python -m pip install . -vv` does have the type `{{ python }}` JINJA template which for some reason fails.

**Comment by carlodri** [comment]
> This seems to work, biggest concern at the moment is that `script: python -m pip install . -vv` does have the type `{{ python }}` JINJA template which for some reason fails.

@conda-forge/core should we be concerned about this?

**Comment by dgasmith** [comment]
@conda-forge/python A gentle poke here if anyone has ideas.

**Comment by carlodri** [comment]
> This seems to work, biggest concern at the moment is that `script: python -m pip install . -vv` does have the type `{{ python }}` JINJA template which for some reason fails.

@conda-forge/core friendly ping :wink: (should we be concerned about this?)

**Comment by marcelotrevisani** [comment]
It is recommended to use ``{{ PYTHON }}`` when calling python for conda recipes

**Comment by marcelotrevisani** [comment]
when you use ``outputs`` you will need to use a bash script
You can see an example here
https://github.com/conda-forge/grayskull-feedstock/blob/master/recipe/meta.yaml
https://github.com/conda-forge/grayskull-feedstock/blob/master/recipe/build_base.sh

**Comment by dgasmith** [comment]
@marcelotrevisani Perfect, thanks!

@carlodri Read to review, FYI I am happy to serve as a maintainer here if you wish.

**Comment by carlodri** [comment]
@conda-forge-admin, please rerender

**Comment by carlodri** [comment]
@conda-forge-admin, please rerender

**Comment by carlodri** [comment]
@conda-forge-admin, please rerender

**Comment by dgasmith** [comment]
Looks good!

**Comment by carlodri** [comment]
Why is conda not picking up the noarch?

**Comment by carlodri** [comment]
Aah probably because of the additional space in

https://github.com/dgasmith/starlette-feedstock/blob/2168c227d6a5311296a97e94aa5d0df6f1727506/recipe/meta.yaml#L35

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,32 +10,48 @@ source:
...
-    - ujson
+   number: 2
+   noarch: python
 
 test:
   commands:
     - python -c "from starlette import *"
 
+outputs:
+  - name: {{ name }}-core
```
Comment (isuruf) [inline recipe/meta.yaml:21]: `base` is more common. For eg: `matplotlib-base`
```suggestion
  - name: {{ name }}-base
```

**Comment by carlodri** [comment]
@conda-forge-admin, please rerender

**Comment by carlodri** [comment]
@conda-forge-admin, please rerender

**Comment by carlodri** [comment]
@isuruf can you help me as to why the `noarch` is not being picked up?

**Comment by carlodri** [comment]
@conda-forge-admin, please rerender

**Pull request merged**